### PR TITLE
Add anonymous instantiate output

### PR DESCRIPTION
**Pull request by guybedford**
To align this format with the changes in 9d88d8929309064b10db24a630419704eb45691f.

Note that while tests have been added, the tests are not actually being executed for some reason. I'm not sure why this is?

### CODE REVIEW

**Comment by johnjbarton** [comment]
The tests from this commit run on Travis and fail
https://travis-ci.org/google/traceur-compiler/builds/50663965

File: src/codegeneration/InstantiateModuleTransformer.js [hunk 1]
```diff
@@ -174,8 +174,11 @@ class InsertBindingAssignmentTransformer extends ScopeTransformer {
  */
 export class InstantiateModuleTransformer extends ModuleTransformer {
 
-  constructor(identifierGenerator) {
-    super(identifierGenerator);
+  constructor(identifierGenerator, reporter, options) {
+    super(identifierGenerator, reporter, options);
+
+    this.anonymousModule =
+      options && !options.bundle && options.moduleName !== true;
```
Comment (arv) [inline src/codegeneration/InstantiateModuleTransformer.js]: options is not optional.

also +2 indentation

File: test/node-api-test.js [hunk 2]
```diff
@@ -165,4 +165,44 @@ suite('node public api', function() {
...
+      module: 'instantiate',
+      moduleName: false
+    }, 'test-module');
+
+    assert.ok(compiled, 'can compile');
+
+    var gotName;
+    var System = {
+      register: function(name) {
+        gotName = typeof name == 'string' && name;
```
Comment (arv) [inline test/node-api-test.js]: ===

File: test/node-api-test.js [hunk 3]
```diff
@@ -165,4 +165,44 @@ suite('node public api', function() {
...
+      }
+    };
+    // eval locally to capture the define() mock
+    eval(compiled);
```
Comment (arv) [inline test/node-api-test.js]: Use global eval maybe?

Comment (arv) [inline test/node-api-test.js]: never mind

ignore moe

**Comment by guybedford** [comment]
I removed the options check but got https://travis-ci.org/google/traceur-compiler/builds/50666590#L207. Have added it back now.

**Comment by guybedford** [comment]
My local copy must have been out of sync that the tests didn't run. All passing now.

File: src/codegeneration/InstantiateModuleTransformer.js [hunk 4]
```diff
@@ -174,8 +174,11 @@ class InsertBindingAssignmentTransformer extends ScopeTransformer {
  */
 export class InstantiateModuleTransformer extends ModuleTransformer {
 
-  constructor(identifierGenerator) {
-    super(identifierGenerator);
+  constructor(identifierGenerator, reporter, options) {
+    super(identifierGenerator, reporter, options);
+
+    this.anonymousModule =
+        options && !options.bundle && options.moduleName !== true;
```
Comment (arv) [inline src/codegeneration/InstantiateModuleTransformer.js]: Can you check the caller?

The short term hack is to mark the option as optional:

```js
constructor(identifierGenerator, reporter, options = undefined) {
```

Comment (guybedford) [inline src/codegeneration/InstantiateModuleTransformer.js]: I've just been copying the pattern in https://github.com/google/traceur-compiler/blob/master/src/codegeneration/AmdTransformer.js#L35.

So you just want to add `undefined` as a default? Isn't that the natural behaviour though?

Comment (arv) [inline src/codegeneration/InstantiateModuleTransformer.js]: We use that as "documentation".

**Comment by guybedford** [comment]
Sure, I've set options to an undefined default.

**Comment by johnjbarton** [comment]
Are we ready here?

**Comment by guybedford** [comment]
Yes, please merge!

File: src/codegeneration/AmdTransformer.js [hunk 5]
```diff
@@ -28,7 +28,7 @@ import scopeContainsThis from './scopeContainsThis.js';
 
 export class AmdTransformer extends ModuleTransformer {
 
-  constructor(identifierGenerator, reporter, options) {
+  constructor(identifierGenerator, reporter, options = undefined) {
```
Comment (johnjbarton) [inline src/codegeneration/AmdTransformer.js]: Why is this change needed?

File: src/codegeneration/CommonJsModuleTransformer.js [hunk 6]
```diff
@@ -43,7 +43,7 @@ import {prependStatements} from './PrependStatements.js';
 
 export class CommonJsModuleTransformer extends ModuleTransformer {
 
-  constructor(identifierGenerator, reporter, options) {
+  constructor(identifierGenerator, reporter, options = undefined) {
```
Comment (johnjbarton) [inline src/codegeneration/CommonJsModuleTransformer.js]: ditto?

Comment (arv) [inline src/codegeneration/CommonJsModuleTransformer.js]: I asked for this. I would have preferred that we tracked down the invalid caller but I did not want to put that burden on @guybedford

File: src/codegeneration/InstantiateModuleTransformer.js [hunk 7]
```diff
@@ -174,8 +174,11 @@ class InsertBindingAssignmentTransformer extends ScopeTransformer {
  */
 export class InstantiateModuleTransformer extends ModuleTransformer {
 
-  constructor(identifierGenerator) {
-    super(identifierGenerator);
+  constructor(identifierGenerator, reporter, options = undefined) {
+    super(identifierGenerator, reporter, options);
+
+    this.anonymousModule =
+        options && !options.bundle && options.moduleName !== true;
```
Comment (arv) [inline src/codegeneration/InstantiateModuleTransformer.js]: As you can see here there is a check for `options`.

File: src/codegeneration/CommonJsModuleTransformer.js (hunk 6 above)
Comment (johnjbarton) [inline src/codegeneration/CommonJsModuleTransformer.js]: The AmdTransformer is created w/o options in one test: https://github.com/google/traceur-compiler/blob/master/test/unit/codegeneration/AmdTransformer.js#L30

I think the change is fine.

**Comment by guybedford** [comment]
Thanks.
